server/tpm/js_ek_tpm_windows.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package tpm
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+	"github.com/nats-io/nkeys"
+)
+
+var (
+	JsKeyTPMVersion    = 1  // Version of the NATS TPM JS implmentation
+	JsKeyTPMDefaultPcr = 22 // Default PCR value for the TPM
+)
+
+// How this works:
+// Create a Storage Root Key (SRK) in the TPM.
+// If existing JS Encrpytion keys do not exist on disk.
+// 	  - Create a JetStream encryption key (js key) and seal it to the SRK
+//      using a provided js encryption key password.
+// 	  - Save the public and private blobs to a file on disk.
+//    - Return the new js encrpytion key (the private portion of the nkey)
+// Otherwise (keys exist on disk)
+//    - Read the public and private blobs from disk
+//    - Load them into the TPM
+//    - Unseal the js key using the TPM, and the provided js encryption keys password.
+//
+// Note: a SRK password for the SRK is supported but not tested here.
+
+// Gets/Regenerates the Storage Root Key (SRK) from the TPM. Caller MUST flush this handle when done.
+func regenerateSRK(rwc io.ReadWriteCloser, srkPassword string) (tpmutil.Handle, error) {
+	// Default EK template defined in:
+	// https://trustedcomputinggroup.org/wp-content/uploads/Credential_Profile_EK_V2.0_R14_published.pdf
+	// Shared SRK template based off of EK template and specified in:
+	// https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-v2.0-Provisioning-Guidance-Published-v1r1.pdf
+	srkTemplate := tpm2.Public{
+		Type:       tpm2.AlgRSA,
+		NameAlg:    tpm2.AlgSHA256,
+		Attributes: tpm2.FlagFixedTPM | tpm2.FlagFixedParent | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth | tpm2.FlagRestricted | tpm2.FlagDecrypt | tpm2.FlagNoDA,
+		AuthPolicy: nil,
+		// We must use RSA 2048 for the intel TSS2 stack
+		RSAParameters: &tpm2.RSAParams{
+			Symmetric: &tpm2.SymScheme{
+				Alg:     tpm2.AlgAES,
+				KeyBits: 128,
+				Mode:    tpm2.AlgCFB,
+			},
+			KeyBits:    2048,
+			ModulusRaw: make([]byte, 256),
+		},
+	}
+	// Create the parent key against which to seal the data
+	srkHandle, _, err := tpm2.CreatePrimary(rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", srkPassword, srkTemplate)
+	return srkHandle, err
+}
+
+type natsTPMPersistedKeys struct {

```

Comment:
These comments are the wrong format and are therefore ignored, struct tags should be like this:
```
type natsTPMPersistedKeys struct {
 	Version    int    `json:"version"`
 	PrivateKey []byte `json:"private_key"`
 	PublicKey  []byte `json:"public_key"`
 }
```

Reply:
Bitten by copilot!  Thanks.